packages/vm/src/vm.ts

```diff
@@ -223,13 +223,19 @@ export class VM extends AsyncEventEmitter<VMEvents> {
     // rather than deep copying the original ones since the copy of the `StateManager`
     // inside the EVM and EEI will be different than the `VM` level copy otherwise
     const eeiCopy = new EEI(stateCopy, commonCopy, blockchainCopy)
-    const evmCopy = new EVM({ eei: eeiCopy, common: commonCopy })
+    const evmCopy = new EVM({
+      eei: eeiCopy,
+      common: commonCopy,
+      allowUnlimitedContractSize: (this.evm as any)._allowUnlimitedContractSize,
```

Comment:
We should also add `customPrecompiles` and `customOpcodes` from the `EVMOpts` property list here.

Reply:
okay